source/extensions/common/aws/utility.cc

```diff
@@ -234,19 +234,45 @@ Utility::joinCanonicalHeaderNames(const std::map<std::string, std::string>& cano
 }
 
 std::string Utility::getSTSEndpoint(absl::string_view region) {
-  if (region == "cn-northwest-1" || region == "cn-north-1") {
-    return fmt::format("sts.{}.amazonaws.com.cn", region);
+
+  // Handle the scenario where we are using SigV4A and a region set has been provided
+  // SigV4A region sets are comma separated and may include wildcard
```

Comment:
Can we move this code comment above as function doc ?

Mentioning, how STS endpoint is derived if a single region is provided or if region set is provided then how a single STS endpoint is derived from that region set.